src/browser/transport.js

```diff
@@ -67,24 +69,33 @@ Transport.prototype.postJsonPayload = function (accessToken, options, jsonPayloa
 
   var method = 'POST';
   var url = _.formatUrl(options);
-  _makeZoneRequest(accessToken, url, method, jsonPayload, callback, requestFactory);
+  var transportOptions = _getMakeZoneRequestOptions(options);
+  _makeZoneRequest(accessToken, url, method, jsonPayload, callback, requestFactory, transportOptions);
+}
+
+// Returns the options to be sent to the _makeZoneRequest calls
+function _getMakeZoneRequestOptions(options) {
+  var timeout = options.timeout;
+  return {
+    timeout: timeout
+  }
 }
 
 // Wraps _makeRequest and if Angular 2+ Zone.js is detected, changes scope
 // so Angular change detection isn't triggered on each API call.
 // This is the equivalent of runOutsideAngular().
 //
-function _makeZoneRequest(accessToken, url, method, data, callback, requestFactory) {
+function _makeZoneRequest(accessToken, url, method, data, callback, requestFactory, transportOptions) {
   var gWindow = ((typeof window != 'undefined') && window) || ((typeof self != 'undefined') && self);
   var currentZone = gWindow && gWindow.Zone && gWindow.Zone.current;
 
   if (currentZone && currentZone._name === 'angular') {
     var rootZone = currentZone._parent;
     rootZone.run(function () {
-      _makeRequest(accessToken, url, method, data, callback, requestFactory);
+      _makeRequest(accessToken, url, method, data, callback, requestFactory, transportOptions);
```

Comment:
The original code here and on line 98 below would have been better as `_makeRequest(arguments);` as it is just passing through the arguments as is. Since we're touching this code, now would be a good time to make this improvement.

Reply:
@waltjones Maybe you prefer doing this:

![image](https://user-images.githubusercontent.com/7487579/119486724-dc8e0a00-bd58-11eb-83d2-daad46cdb7b3.png)

Is it ok for you adding "this" for the thisArg? I've checked your source code and it seems "this" is valid although I don't know if you prefer "undefined" ?

Reply:
Yes that's what I intended. Thank you for correcting that.

`this` won't reference `Transport` here, so I think using undefined makes the intent more clear.

Reply:
Perfect, applied changes